making_change/making_change.py

```diff
@@ -3,7 +3,17 @@
 import sys
 
 def making_change(amount, denominations):
```

Comment:
The logic for this problem was provided in the instructions. Here is my solution:

```
import sys

def making_change(amount, denominations):
  cache = { 0: 1 }

  for a in range(denominations[0], amount + 1):
    cache[a] = cache[a - denominations[0]]

  for b in range(denominations[1], amount + 1):
    cache[b] = cache[b] + cache[b - denominations[1]]

  for c in range(denominations[2], amount + 1):
    cache[c] = cache[c] + cache[c - denominations[2]]

  for d in range(denominations[3], amount + 1):
    cache[d] = cache[d] + cache[d - denominations[3]]

  for e in range(denominations[4], amount + 1):
    cache[e] = cache[e] + cache[e - denominations[4]]
  
  return cache[amount] 
```